rudder-web/src/main/scala/com/normation/rudder/web/rest/node/NodeAPIService8.scala

```diff
@@ -81,4 +90,66 @@ class NodeApiService8 (
       saved
     }
   }
+
+ private[this] val pipeSize = 4096
+
+  def runResponse(in : InputStream)(out : OutputStream) = {
+    val bytes : Array[Byte] = new Array(pipeSize)
+    val zero = 0.toByte
+    var read = 0
+    try {
+      while (read != -1) {
+        Arrays.fill(bytes,zero)
+        read = in.read(bytes)
+        out.write(bytes)
+        out.flush()
+      }
+
+    } catch {
+      case e : IOException =>
+        // should we log ? Should we end ?
```

Comment:
You can't do that, or you will leak ressource. The best way to handle that is to let the exception throws here, and manages it above. As your method is only used locally, it's ok. So remove try/catch here.